projects/cds/src/profiletag/services/profile-tag.injector.ts

```diff
@@ -93,6 +103,29 @@ export class ProfileTagInjector {
       );
   }
 
+  /**
+   * Listens to the changes to the cart and pushes the event for profiletag to pick it up further.
+   */
+  private cartChanged(): Observable<boolean> {
+    return combineLatest([
+      this.cartService.getEntries(),
+      this.cartService.getActive(),
+    ]).pipe(
+      skipWhile(([entries]) => entries.length === 0),
+      tap(([entries, cart]) => {
+        this.notifyProfileTagOfCartChange({ entries, cart });
+      }),
+      mapTo(true)
+    );
+  }
+
+  private notifyProfileTagOfCartChange({ entries, cart }): void {
```

Comment:
There's no type safety for `entries` and `cart`. If you hover over them in VSCode, it will say `any`.
To add type safety and keep destructuring, use this syntax: `{ entries, cart }: {entries: OrderEntry[], cart: Cart}`.

Reply:
The same for `notifyProfileTagOfConsentChange()` method